source-gallery/src/main/java/com/google/android/apps/muzei/gallery/GalleryImportPhotosDialogFragment.java

```diff
@@ -0,0 +1,100 @@
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.widget.ArrayAdapter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GalleryImportPhotosDialogFragment extends DialogFragment {
+    public static final String TAG = GalleryImportPhotosDialogFragment.class.getSimpleName();
+
+    private List<ActivityInfo> mGetContentActivities = new ArrayList<>();
+    private OnRequestContentListener mListener;
+    private ArrayAdapter<CharSequence> mAdapter;
+
+    public static GalleryImportPhotosDialogFragment createInstance() {
+        return new GalleryImportPhotosDialogFragment();
+    }
+
+    public GalleryImportPhotosDialogFragment () {
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mAdapter = new ArrayAdapter<>(getSafeContext(), android.R.layout.select_dialog_item);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        return new AlertDialog.Builder(getSafeContext())
+                .setTitle(R.string.gallery_import_dialog_title)
+                .setAdapter(mAdapter, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        mListener.requestGetContent(mGetContentActivities.get(which));
+                    }
+                }).create();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnRequestContentListener) {
+            mListener = (OnRequestContentListener) context;
+        } else {
+            throw new IllegalArgumentException(context.getClass().getSimpleName() + " must implement OnRequestContentListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mGetContentActivities = mListener.getGetContentActivityInfoList();
+        if (mAdapter == null || mGetContentActivities == null || mGetContentActivities.isEmpty()) {
+            this.dismiss();
+            return;
+        }
+
+        updateAdapter();
+    }
+
+    private void updateAdapter() {
+        PackageManager packageManager = getSafeContext().getPackageManager();
+        int size = mGetContentActivities.size();
+        List<CharSequence> items = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            items.add(mGetContentActivities.get(i).loadLabel(packageManager));
+        }
+        mAdapter.clear();
+        mAdapter.addAll(items);
+        mAdapter.notifyDataSetChanged();
+    }
+
+    private Context getSafeContext() {
```

Comment:
This method is unnecessary with Support Fragments. Just use `getContext()`

Reply:
Sounds good. I did not know that

Reply:
So it will only cause a problem if I use the Vanilla Fragment?

Reply:
Well yeah, for framework Fragments, `getContext` didn't exist until API 23 as per [the documentation](https://developer.android.com/reference/android/app/Fragment.html#getContext())

Reply:
Sorry for the dumb question :(